### PR TITLE
Stomp action

### PR DESCRIPTION
**Pull request by Gavin-L26**
Stomp Action added with tutorial stomp FX added to level one scene to see if it works. It would be better to test with other level's midi that has stomp action in it.

- All action are now triggered through unity input system callback only. The keyCode check for the keyboard is removed.

### CODE REVIEW

File: Assets/Scenes/LevelOneScene.unity [hunk 1]
```diff
@@ -330,7 +330,7 @@ Transform:
   m_ConstrainProportionsScale: 0
```
Comment (jamilfaisal) [inline Assets/Scenes/LevelOneScene.unity:330]: Can you apply these changes to the level two scene as well?

File: Assets/Scripts/PlayerScripts/PlayerMovement.cs [hunk 2]
```diff
@@ -278,6 +241,16 @@ public void TriggerJump(InputAction.CallbackContext context)
             }
         }
     }
+
+    public void TriggerStomp(InputAction.CallbackContext context)
+    {
+        if (!_playerInputEnabled) return;
+
+        if (!context.canceled){
```
Comment (jamilfaisal) [inline Assets/Scripts/PlayerScripts/PlayerMovement.cs]: Will the stomp sound and stomp method be triggered twice? (Once for context.started and once for context.performed?)

File: Assets/Scenes/LevelOneScene.unity (hunk 1 above)
Comment (Gavin-L26) [reply Assets/Scenes/LevelOneScene.unity:330]: yep will do

File: Assets/Scripts/PlayerScripts/PlayerMovement.cs (hunk 2 above)
Comment (Gavin-L26) [reply Assets/Scripts/PlayerScripts/PlayerMovement.cs]: From my test, checking started and performed doesn't actually prevent it from triggering twice (once for press once for release), only checking cancelled worked

Comment (Gavin-L26) [reply Assets/Scripts/PlayerScripts/PlayerMovement.cs]: Did you hear that it triggered twice?

Comment (jamilfaisal) [reply Assets/Scripts/PlayerScripts/PlayerMovement.cs]: I put a log statement inside the if-statement. Seems to be triggering twice
image

Remember, these methods get called 3 times, once for context.started, once for context.performed, and once for context.canceled. Should probably just replace if(!context.canceled) with if (context.performed)

**Comment by jamilfaisal** [comment]
The player can stomp when they're on the ground. I think that "if (!context.canceled)" needs another boolean for checking if the player is not grounded

**Comment by Gavin-L26** [comment]
> The player can stomp when they're on the ground. I think that "if (!context.canceled)" needs another boolean for checking if the player is not grounded

hmmm ok, does that mean if a stomp command is right after a jump, if the player did not jump then they cannot trigger the stomp after it too?

**Comment by jamilfaisal** [comment]
The stomp feels wayyyyyyyy too weak. I increased stomp force to 40 and it seems a lot better (although this change was made while the stomp triggers twice so you may have to increase that value even more, once the double method trigger is fixed).

**Comment by Gavin-L26** [comment]
> The stomp feels wayyyyyyyy too weak. I increased stomp force to 40 and it seems a lot better (although this change was made while the stomp triggers twice so you may have to increase that value even more, once the double method trigger is fixed).

Sure I didn't change stomp value so it's the same one as before, but I can change it now

**Comment by jamilfaisal** [comment]
> > The player can stomp when they're on the ground. I think that "if (!context.canceled)" needs another boolean for checking if the player is not grounded
> 
> hmmm ok, does that mean if a stomp command is right after a jump, if the player did not jump then they cannot trigger the stomp after it too?

Sure that's one way to do it. I thought we already had written code for only letting the player stomp if they're in midair?
Right, this is it:
image

Ohhh, you're playing the sound before the check for whether the player is on the ground lol. Just move the "stompSound.Play();" to after the "if (_grounded) return;" check (or move the boolean !_grounded to the if(!context.canceled) check)

**Comment by Gavin-L26** [comment]
> 

Yeah I can fix the sound part, I was more asking about the player action trigger. We still want the player to be able to press the stomp button for triggering its score calculation even when they are on ground right? I mean that feels more fair to me

**Comment by Gavin-L26** [comment]
> > > The player can stomp when they're on the ground. I think that "if (!context.canceled)" needs another boolean for checking if the player is not grounded
> > 
> > 
> > hmmm ok, does that mean if a stomp command is right after a jump, if the player did not jump then they cannot trigger the stomp after it too?
> 
> Sure that's one way to do it. I thought we already had written code for only letting the player stomp if they're in midair? Right, this is it: image
> 
> Ohhh, you're playing the sound before the check for whether the player is on the ground lol. Just move the "stompSound.Play();" to after the "if (_grounded) return;" check (or move the boolean !_grounded to the if(!context.canceled) check)

Oh I see, yeah that can be a quick fix. But I was more asking about the player action trigger. We still want the player to be able to press the stomp button for triggering its score calculation even when they are on ground right? I mean that feels more fair to me

**Comment by jamilfaisal** [comment]
> > > > The player can stomp when they're on the ground. I think that "if (!context.canceled)" needs another boolean for checking if the player is not grounded
> > > 
> > > 
> > > hmmm ok, does that mean if a stomp command is right after a jump, if the player did not jump then they cannot trigger the stomp after it too?
> > 
> > 
> > Sure that's one way to do it. I thought we already had written code for only letting the player stomp if they're in midair? Right, this is it: image
> > Ohhh, you're playing the sound before the check for whether the player is on the ground lol. Just move the "stompSound.Play();" to after the "if (_grounded) return;" check (or move the boolean !_grounded to the if(!context.canceled) check)
> 
> Oh I see, yeah that can be a quick fix. But I was more asking about the player action trigger. We still want the player to be able to press the stomp button for triggering its score calculation even when they are on ground right? I mean that feels more fair to me

Oh yeah I agree. So that if they stomp while on the ground, they lose points for it. Good idea

**Comment by Gavin-L26** [comment]
> The stomp feels wayyyyyyyy too weak. I increased stomp force to 40 and it seems a lot better (although this change was made while the stomp triggers twice so you may have to increase that value even more, once the double method trigger is fixed).

Should we instead apply a constant force for stomping until they hit ground? Since rn it only apply it for one frame when it is called, which is probably why it can feel very slow.

**Comment by jamilfaisal** [comment]
> > The stomp feels wayyyyyyyy too weak. I increased stomp force to 40 and it seems a lot better (although this change was made while the stomp triggers twice so you may have to increase that value even more, once the double method trigger is fixed).
> 
> Should we instead apply a constant force for stomping until they hit ground? Since rn it only apply it for one frame when it is called, which is probably why it can feel very slow.

Yeah that's one of the polish tasks actually. If you got time to do it then sure. I remember testing with 40 stomp force felt like a strong push downwards, which was fine because it got me to the ground really fast

**Comment by jamilfaisal** [comment]
The stomp sound effect is not affected by the sound effects volume set in the settings menu. You need to add the stomp sound effect to SoundVolumeManager.cs

**Review by jamilfaisal — APPROVED** [review]
Good work! All that's left is doing the changes for level two scene

**Review by lilvedaes — COMMENTED** [review]
Stomp feels a bit too quiet to me, the clap sound is supposed to be very loud. Besides that, you can test in the tutorial level if you want since that one has a couple stomps c:

**Review by lilvedaes — CHANGES_REQUESTED** [review]
Can we make stomp louder for tutorial level and level 1?
Also, is the stomp sound the same for tutorial and level one? If not we should probably change it

**Comment by Gavin-L26** [comment]
> Can we make stomp louder for tutorial level and level 1? Also, is the stomp sound the same for tutorial and level one? If not we should probably change it

Just for stomp and just for tutorial level and level one? I think the original sound FX for the tutorial stomp is just much quieter than any other sound FX. I guess I have to add public variable on SoundVolumeManger, so we can have scene-dependant individual sound FX volume

**Comment by lilvedaes** [comment]
> > Can we make stomp louder for tutorial level and level 1? Also, is the stomp sound the same for tutorial and level one? If not we should probably change it
> 
> Just for stomp and just for tutorial level and level one? I think the original sound FX for the tutorial stomp is just much quieter than any other sound FX. I guess I have to add public variable on SoundVolumeManger, so we can have scene-dependant individual sound FX volume

the stomp sound is good on level two. The sound is the same for tutorial and level one, both have the clap sound, is this intended? If not that has to change c: That clap sound is also not loud enough, can you make it louder?

**Pull request merged**